File: GeeksForGeeks/Arrays/WaveSort.cpp
#include "iostream"
using namespace std;
int main(int argc, char* argv[])
{

    int arr[] = {1, 3, 4, 2, 7, 8};
    int n = sizeof(arr)/sizeof(arr[0]);
    for(int i=0;i<n;i+=2) {
        if(i>0 and arr[i-1]>arr[i]) {
            swap(arr[i-1], arr[i]);
        }
        if(i<=n-2 and arr[i+1]>arr[i]) {
            swap(arr[i+1], arr[i]);
        }
    }
    for(int i=0;i<n;i++) {
        cout<<arr[i]<<" ";
    }

    return 0;
}

File: UnacademyCourse/SortingAndSearchingPythonJune2021/Lecture2/WaveSort.py
def wave_sort(arr):
    # TC - O(n) SC - O(1)
    i = 0
    n = len(arr)
    while i < n:
        if i > 0 and arr[i-1] > arr[i]:
            arr[i-1], arr[i] = arr[i], arr[i-1]
        if i < n-1 and arr[i] < arr[i+1]:
            arr[i+1], arr[i] = arr[i], arr[i+1]
        i += 2
    
arr = [5,2,6,7,2,1,0,3]
wave_sort(arr)
print(arr)

